src/pages/workspaces/workspace/analysis/modals/AnalysisModal.ts

```diff
@@ -312,18 +360,19 @@ export const AnalysisModal = withDisplayName('AnalysisModal')(
         ]),
         div({ style: { display: 'flex', justifyContent: 'flex-end', marginTop: '1rem' } }, [
           h(ButtonPrimary, {
-            disabled: errors,
+            //TODO: See spinner overlay comment. Change to pendingCreate || errors.
+            disabled: status === 'Loading' || pendingCreate || errors,
             tooltip: Utils.summarizeErrors(errors),
             onClick: async () => {
               try {
                 const contents = Utils.cond(
                   [isJupyterLab || isJupyter, () => JSON.stringify(notebookData[notebookKernel])],
                   [isRStudio, () => baseRmd])
                 const fullAnalysisName = `${analysisName}.${fileExt}`
-                !!googleProject ?
-                  await Ajax().Buckets.analysis(googleProject, bucketName, fullAnalysisName, toolLabel).create(contents) :
-                  await Ajax().AzureStorage.blob(workspaceId, fullAnalysisName).create(contents)
+                await create(fullAnalysisName, toolLabel, contents)
+                //TODO: Temporary, once Analyses.js uses store, refreshAnalyses will be deprecated in favor of refresh() within the create function
                 await refreshAnalyses()
+                // @ts-expect-error
                 await Ajax().Metrics.captureEvent(Events.analysisCreate, { source: toolLabel, application: toolLabel, filename: fullAnalysisName })
```

Comment:
I have a useMetrics hook that will make this less awkward. (part of my pending PR)